viscy/light/engine.py

```diff
@@ -305,8 +323,16 @@ def on_train_epoch_end(self):
         self.training_step_outputs = []
 
     def on_validation_epoch_end(self):
+        super().on_validation_epoch_end()
         self._log_samples("val_samples", self.validation_step_outputs)
         self.validation_step_outputs = []
+        # average within each dataloader
```

Comment:
Here I know that this is the end of the validation meaning that it won't backprop and therefore maybe no need to detach the tensor before doing any logging. Is this the common practice? I know that for the `train_step` and `validation_step` is more relevant and important If we don't detach here it doesnt affect it any way? Just curious..

Reply:
To log the loss value (through lightning logger), the detaching is automatic. We only need to care about it when logging manually (images).